File: 162. Find Peak Element/162. Find Peak Element.cpp
class Solution
{
public:
    int findPeakElement(vector<int> &nums)
    {
        vector<int> v;
        v.push_back(INT_MIN);
        int n = nums.size();

        if (n == 1)
        {
            if (nums[0] == INT_MIN)
            {
                return 0;
            }
            else if (nums[0] == INT_MAX)
            {
                return 0;
            }
        }

        for (int i = 0; i < n; i++)
        {
            v.push_back(nums[i]);
        }

        v.push_back(INT_MIN);

        int fix = 0;

        int N = v.size();

        for (int i = 1; i <= N - 2; i++)
        {
            if ((v[i] > v[i - 1]) && (v[i] > v[i + 1]))
            {
                fix = i;
                break;
            }
        }

        return fix - 1;
    }
};

File: 162. Find Peak Element/162. Find Peak Element.java
public int findPeakElement(int[] nums) {
    int N = nums.length;
    if (N == 1) {
        return 0;
    }
   
    int left = 0, right = N - 1;
    while (right - left > 1) {
        int mid = left + (right - left) / 2;
        if (nums[mid] < nums[mid + 1]) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    
    return (left == N - 1 || nums[left] > nums[left + 1]) ? left : right;
}